webapp-client/src/component/CategoryList.js

```diff
@@ -42,8 +42,10 @@ export default class CategoryList extends React.Component {
     this.getCategoryList();
   }
 
-  componentDidUpdate() {
-    this.getCategoryList();
+  componentDidUpdate(prevProps) {
+    if (prevProps !== this.props) {
```

Comment:
I also changed this in another branch (delete-items, im v sorry accidentally i wrote to it thinking it was delete-category), but because the state also changes, so it should be (prevProps, prevState) and then if(prevProps !== this.props || prevState !== this.state)